prow/crier/reporters/pubsub/reporter.go

```diff
@@ -98,12 +98,16 @@ func (c *Client) Report(_ *logrus.Entry, pj *prowapi.ProwJob) ([]*prowapi.ProwJo
 	message := c.generateMessageFromPJ(pj)
 
 	ctx := context.Background()
+	// TODO: Consider caching the pubsub client.
 	client, err := pubsub.NewClient(ctx, message.Project)
```

Comment:
The client also has a `Close()`, mostly to shutdown its connection pool apparently, maybe `defer` that as well?

Reply:
Yes, I'm already (sneakily) doing that on line 107 within the log statement. 😄

Reply:
Ah, missed that. I am pretty sure this doesn't do what you think though, the log statement itself does get deferred, but the resolving of its argument does not, hence you will close the client in line 107 and not via the `defer`

Reply:
Good catch, I thought the arguments were evaluated lazily. Fixed.